database/migrations/000093_profile_status_trigger.up.sql

```diff
@@ -0,0 +1,203 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Start to make sure the function and trigger are either both added or none
+BEGIN;
+
+-- drop the old triggers based on rule_details_eval and rule_evaluations, and replace with latest_evaluation_statuses
+-- (See migrations #7 and #54)
+DROP TRIGGER IF EXISTS update_profile_status_after_delete ON rule_evaluations;
+DROP TRIGGER IF EXISTS update_profile_status ON rule_details_eval;
+
+-- Trigger function for updates
+CREATE OR REPLACE FUNCTION update_profile_status() RETURNS TRIGGER AS $$
+DECLARE
+    v_status eval_status_types;
+    v_new_status eval_status_types;
+    v_other_error boolean;
+    v_other_failed boolean;
+    v_other_success boolean;
+    v_other_skipped boolean;
+    v_pending boolean;
+BEGIN
+  -- Fetch the profile_id for the new evaluation status
+  SELECT es.status INTO v_new_status
+  FROM latest_evaluation_statuses AS les
+  JOIN evaluation_statuses AS es ON es.id = les.evaluation_history_id
+  WHERE les.profile_id = NEW.profile_id
+  AND les.rule_entity_id = NEW.rule_entity_id;
+
+  IF v_new_status IS NULL THEN
+      RAISE EXCEPTION 'oh no';
```

Comment:
While this gave me a chuckle, perhaps we should use a more descriptive exception message?

Reply:
This was a debugging message I forgot to remove